{"id": 29510, "wybug_id": "wooyun-2015-0119116", "wybug_title": "苏宁APP 苏宁应用商店&amp;易付宝钱包拒绝服务打包", "wybug_corp": "江苏苏宁易购电子商务有限公司", "wybug_author": "小手冰凉", "wybug_date": "2015-06-09 15:45", "wybug_open_date": "2015-09-07 16:28", "wybug_type": "拒绝服务", "wybug_level": "中", "wybug_rank_0": "10", "wybug_status": "厂商已经确认", "wybug_from": "http://www.wooyun.org", "wybug_tags": ["拒绝服务"], "wybug_detail": "漏洞详情 披露状态：   \t\t\t\t\t\t\t\t\t2015-06-09：\t细节已通知厂商并且等待厂商处理中\t\t\t\t\t\t\t\t\t2015-06-09：\t厂商已经确认，细节仅向厂商公开\t\t\t\t\t\t\t\t\t2015-06-12：\t细节向第三方安全合作伙伴开放\t\t\t\t\t\t\t\t\t2015-08-03：\t细节向核心白帽子及相关领域专家公开\t\t\t\t\t\t\t\t\t2015-08-13：\t细节向普通白帽子公开\t\t\t\t\t\t\t\t\t2015-08-23：\t细节向实习白帽子公开\t\t\t\t\t\t\t\t\t2015-09-07：\t细节向公众公开  简要描述： 苏宁应用商店&易付宝钱包拒绝服务 详细说明：  第一处：receiver：com.suning.market.core.broadcast.PackageReceiver测试命令\nadb shell am broadcast -a android.intent.action.PACKAGE_ADDED -c android.intent.category.HOME -n com.suning.market/com.suning.market.core.broadcast.PackageReceiver\n该receiver中onReceive方法调getDataString未检测\n\n第二处：activity：com.suning.mobile.paysdk.ui.CashierPrepareActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.CashierPrepareActivity\n应该是测试用的接口，苏宁易购app中也有这个，原因一样其中发生了如下调用关系\ncom.suning.mobile.paysdk.ui.CashierPrepareActivity.onCreate()com.suning.mobile.paysdk.ui.CashierPrepareActivity.c()com.suning.mobile.paysdk.ui.c.a.a.a()com.suning.mobile.paysdk.ui.c.a.a.b()\n最后的函数b有如下未验证代码导致程序崩溃\n\n第三处：activity：com.suning.mobile.paysdk.ui.activity.CashierChannelActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.activity.CashierChannelActivity\n这里最终引发异常的是at com.suning.mobile.paysdk.ui.ba这个类的c方法，由于混淆比较严重不能确定出问题的具体代码，但是根据里面的代码能出现空指针的只能是其中的变量未初始化。第四处：activity：com.suning.mobile.paysdk.ui.QPayFirstActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.QPayFirstActivity\n这里出问题的跟上面是同一个位置第五处：activity：com.suning.mobile.paysdk.ui.BankListActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.BankListActivity\n引发异常的是com.suning.mobile.paysdk.ui.a类中的c方法，函数中有可能引发异常的地方就是本地变量未初始化。第六处：activity：com.suning.mobile.paysdk.ui.SMSCheckActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.SMSCheckActivity\n引发异常的是com.suning.mobile.paysdk.ui.bg类的onCreate方法。其中使用了未检查的方法。\n\n第七处：activity：com.suning.mobile.paysdk.ui.QPayProtocolActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.QPayProtocolActivity\nActivity能够成功打开，但是同时activity同时启动一个线程com.suning.mobile.paysdk.ui.ay 这个线程中出现未初始化变量导致异常第八处：activity：com.suning.mobile.paysdk.ui.PaySuccessActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.PaySuccessActivity\n引发异常的是com.suning.mobile.paysdk.ui.aa的b方法。在界面启动时控件没有初始化，但是该方法就已经在操作控件导致空指针，从界面上观察可看出。第九出：activity：com.suning.mobile.paysdk.ui.EppActivateActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.ui.EppActivateActivity\nEppActivateActivity的onCreate方法调用了未检查的getString方法。\n\n第十出：activity：com.suning.mobile.paysdk.pay.cashierpay.CashierPrepareActivity测试命令\nadb shell am start -n com.suning.mobile.epa/com.suning.mobile.paysdk.pay.cashierpay.CashierPrepareActivity\n经过多层调用，异常最终发生在com.suning.mobile.paysdk.pay.cashierpay.c.a的a方法，该方法调用了未检查的getStringArray方法。\n\n   漏洞证明：  崩溃效果都一样，上一张图\n\n   修复方案：  除了注意常规的get方法和变量初始化外，在release版本中应当将测试接口全部关闭，只要在AndroidManifest.xml中删除就行了，不然一般测试用的接口都故意做的非常容易调用，被恶意程序利用更加方便。   版权声明：转载请注明来源 小手冰凉@乌云\n ", "wybug_reply": "漏洞回应  厂商回应： 危害等级：中 漏洞Rank：10  确认时间：2015-06-09 16:27 厂商回复： 感谢提交，移动客户端拒绝服务漏洞一律按照200元礼品卡计算。 最新状态： 暂无  ", "replys": "漏洞评价：\n评论\n     2015-06-11 01:54 |    \t\t板凳 \t\t\t( 路人 |\t\t\t        Rank:0 漏洞数:1        | http://www.bandeng.cc)\t\t \n  楼主 是否可发个email给我 6970766QQ    \n  \n\n\n", "wybug_level_fromcorp": "中", "wybug_rank_fromcorp": 10, "Ranks": null}